EA-2183: Add search description to manifests

ikattey opened the pull request.

ikattey committed:
EA-2183: Add service description to manifests

- made type property nullable in v3 service descriptions
- configured Jackson to handle Java 8 optionals

P-Ehlert commented on src/main/java/eu/europeana/iiif/model/v2/ManifestV2.java:
```diff
@@ -12,12 +12,13 @@
  * Created on 06-02-2018
  */
-@JsonPropertyOrder({"id", "context"})
+@JsonPropertyOrder({"id", "service", "context"})
```
This should probably read {"id", "service", "@ context"} (without a space beween @ and context, but that was added to avoid refering to a git-user named context).  However I don't like that order.
I just talked to the PM and he prefers to keep the same order defined in his EDM mapping document. However that does not have service yet, so I asked him to add this.

ikattey replied:
Remarks now implemented.

ikattey committed:
Update service description in manifests 

- service endpoint uses fulltextApi config setting, instead of hard-coded string
- updated service path
- changed service positioning in manifest

ikattey committed:
Remove unused field

ikattey committed:
Optimise fullTextApiBaseUrl lookup when creating search ID
